lib/chef/knife/vsphere_vm_clone.rb

```diff
@@ -325,16 +337,15 @@ def run
 
     return if get_config(:mark_as_template)
     if get_config(:power) || get_config(:bootstrap)
-      vm = find_in_folder(dest_folder, RbVmomi::VIM::VirtualMachine, vmname) ||
+      @vm ||= find_in_folder(dest_folder, RbVmomi::VIM::VirtualMachine, vmname) ||
```

Comment:
Did you consider passing `vm` to `connect_host` and `ipv4_address` rather than using an instance var? 

Also the `||=` is a bit jarring because it shouldn't be possible to get here with `@vm` set, but the use of `||=` indicates it might be.